docs/ldml/tr35-keyboards.md

```diff
@@ -1378,30 +1380,149 @@ This is a container for variables to be used with transforms.
+<key id="acute_key" to="${acute}" />
+…
+<!-- Display ´ instead of the non-displayable marker -->
+<display to="${acute}" display="´" />
+```
+
+* * *
+
+### Element: set
+
+> <small>
+>
+> Parents: [variables](#Element_variables)
+>
+> Children: _none_
+>
+> Occurrence: optional, multiple
+> </small>
+
+> This element represents a set of strings used by the [transform](#element-transform) elements for string matching and substitution.
+
+_Attribute:_ `id` (required)
+
+> Specifies the identifier (name) of this set.
+> All ids must be unique across all types of variables.
+
+_Attribute:_ `value` (required)
+
+> The `value` attribute is always a set of strings separated by whitespace, even if there is only a single item in the set, such as `"A"`.
+> Leading and trailing whitespace is ignored.
+> This attribute may be escaped with `\u` notation, see [Escaping](#Escaping).
+> Sets may refer to other string variables if they have been previously defined, using `${string}` syntax, or to other previously-defined sets using `$[set]` syntax.
+> [Markers](#markers) may be included with the `\m{…}` notation.
+
+**Examples**
+
+```xml
+<variables>
+    <set id="upper" value="A B CC D E FF " /> <!-- 6 items -->
+    <set id="lower" value="a b c  d e  f " /> <!-- 6 items -->
+    <set id="upper_or_lower" value="$[upper] $[lower]"  /> <!-- Concatenate two sets -->
+    <set id="lower_or_upper" value="$[lower] $[upper]"  /> <!-- Concatenate two sets -->
+    <set id="a" value="A"/> <!-- Just one element, an 'A'-->
+    <set id="cluster_or_zwnj" value="${hi_cluster} ${zwnj}"/> <!-- 2 items: "हि \u${200C}"-->
+</variables>
+```
+
+Match "X" followed by any uppercase letter:
+
+```xml
+<transform from="X$[upper]" to="…" />
+```
+
+Map from upper to lower:
 
+```xml
+<transform from="($[upper])" to="$[1:lower]" />
 ```
 
-See the `from=` and `to=` sections of [`transforms`](#Element_transforms) for details.
+See [transform](#element-transform) for further details.
+
+* * *
+
+### Element: unicodeSet
+
+> <small>
+>
+> Parents: [variables](#Element_variables)
+>
+> Children: _none_
+>
+> Occurrence: optional, multiple
+> </small>
+
+> This element represents a set, using a subset of the [UnicodeSet](tr35.md#Unicode_Sets) format, used by the [transform](#element-transform) elements for string matching and substitution.
+> Note important restrictions on the syntax below.
+
+_Attribute:_ `id` (required)
+
+> Specifies the identifier (name) of this unicodeSet.
+> All ids must be unique across all types of variables.
+
+_Attribute:_ `value` (required)
+
+> String value in [UnicodeSet](tr35.md#Unicode_Sets) format.
+> Leading and trailing whitespace is ignored.
+> Variables may refer to other string variables if they have been previously defined, using `${string}` syntax, or to other previously-defined UnicodeSets (not sets) using `$[unicodeSet]` syntax.
+
+**Syntax Note**
+
+- Multi-character strings (`{}`) are not supported, such as `[żġħ{ie}{għ}]`.
```

Comment:
```suggestion
- Warning: UnicodeSets look superficially similar to regex character classes as used in [`transform`](#element-transform) elements, but they are different. UnicodeSets must be defined with a `unicodeSet` element, and referenced with the `$[unicodeSet]` notation in transforms. UnicodeSets cannot be used directly in a transform.
- Multi-character strings (`{}`) are not supported, such as `[żġħ{ie}{għ}]`.
```

I'm not sure of the wording, but I can see keyboard devs tripping over on this (and it might be good to include similar wording under the transform element).

Reply:
or at least "see #element-unicodeSet" from there. Agreed. Also this is a good reason to NOT have UnicodeSet intermingled in the regex syntax.